UPGRADE-3.x.md

```diff
@@ -32,6 +32,19 @@ extend the corresponding classes in `SonataAdminBundle\Action\`.
 If you need to style headers prefer to use CSS classes and not in the html DOM.
 In this case please use `header_class` option.
 
+## Deprecated return type of SimplePager::getResults
+When calling `SimplePager::getResults()` on non-empty result which extends `$maxPerPage` value, `ArrayCollection` would be returned instead of `array`
+as it is declared in `PagerInterface`. Update usage of `SimplePager::getResults()`, ensure you are transform `Collection` to an `array` and you do not work with `Collection` methods
```

Comment:
```suggestion
## Deprecated returning other type than `ArrayCollection` from `SimplePager::getResults()`

When calling `SimplePager::getResults()` on non-empty result which has set `$maxPerPage`, `ArrayCollection` would be returned instead of `array` as it is declared in `PagerInterface`. Update usage of `SimplePager::getResults()`, ensure you are transforming `Collection` to `array` and you aren't dealing with any of its methods.
```